rpcs3/Gui/padsettingsdialog.cpp

```diff
@@ -104,4 +270,362 @@ PadSettingsDialog::PadSettingsDialog(QWidget *parent) : QDialog(parent)
+	b_down_lstick->setText(GetKeyName(g_kbpad_config.left_stick_down));
+	b_left_lstick->setText(GetKeyName(g_kbpad_config.left_stick_left));
+	b_right_lstick->setText(GetKeyName(g_kbpad_config.left_stick_right));
+	
+	b_up->setText(GetKeyName(g_kbpad_config.up));
+	b_down->setText(GetKeyName(g_kbpad_config.down));
+	b_left->setText(GetKeyName(g_kbpad_config.left));
+	b_right->setText(GetKeyName(g_kbpad_config.right));
+	
+	b_shift_l1->setText(GetKeyName(g_kbpad_config.l1));
+	b_shift_l2->setText(GetKeyName(g_kbpad_config.l2));
+	b_shift_l3->setText(GetKeyName(g_kbpad_config.l3));
+	
+	b_start->setText(GetKeyName(g_kbpad_config.start));
+	b_select->setText(GetKeyName(g_kbpad_config.select));
+	
+	b_shift_r1->setText(GetKeyName(g_kbpad_config.r1));
+	b_shift_r2->setText(GetKeyName(g_kbpad_config.r2));
+	b_shift_r3->setText(GetKeyName(g_kbpad_config.r3));
+	
+	b_square->setText(GetKeyName(g_kbpad_config.square));
+	b_cross->setText(GetKeyName(g_kbpad_config.cross));
+	b_circle->setText(GetKeyName(g_kbpad_config.circle));
+	b_triangle->setText(GetKeyName(g_kbpad_config.triangle));
+	
+	b_up_rstick->setText(GetKeyName(g_kbpad_config.right_stick_up));
+	b_down_rstick->setText(GetKeyName(g_kbpad_config.right_stick_down));
+	b_left_rstick->setText(GetKeyName(g_kbpad_config.right_stick_left));
+	b_right_rstick->setText(GetKeyName(g_kbpad_config.right_stick_right));
+}
+
+void PadSettingsDialog::ResetParameters()
+{
+	g_kbpad_config.left_stick_up = g_kbpad_config.left_stick_up.def;
+	g_kbpad_config.left_stick_down = g_kbpad_config.left_stick_down.def;
+	g_kbpad_config.left_stick_left = g_kbpad_config.left_stick_left.def;
+	g_kbpad_config.left_stick_right = g_kbpad_config.left_stick_right.def;
+	
+	g_kbpad_config.up = g_kbpad_config.up.def;
+	g_kbpad_config.down = g_kbpad_config.down.def;
+	g_kbpad_config.left = g_kbpad_config.left.def;
+	g_kbpad_config.right = g_kbpad_config.right.def;
+	
+	g_kbpad_config.l1 = g_kbpad_config.l1.def;
+	g_kbpad_config.l2 = g_kbpad_config.l2.def;
+	g_kbpad_config.l3 = g_kbpad_config.l3.def;
+	
+	g_kbpad_config.start = g_kbpad_config.start.def;
+	g_kbpad_config.select = g_kbpad_config.select.def;
+	
+	g_kbpad_config.r1 = g_kbpad_config.r1.def;
+	g_kbpad_config.r2 = g_kbpad_config.r2.def;
+	g_kbpad_config.r3 = g_kbpad_config.r3.def;
+	
+	g_kbpad_config.square = g_kbpad_config.square.def;
+	g_kbpad_config.cross = g_kbpad_config.cross.def;
+	g_kbpad_config.circle = g_kbpad_config.circle.def;
+	g_kbpad_config.triangle = g_kbpad_config.triangle.def;
+	
+	g_kbpad_config.right_stick_up = g_kbpad_config.right_stick_up.def;
+	g_kbpad_config.right_stick_down = g_kbpad_config.right_stick_down.def;
+	g_kbpad_config.right_stick_left = g_kbpad_config.right_stick_left.def;
+	g_kbpad_config.right_stick_right = g_kbpad_config.right_stick_right.def;
+}
+
+void PadSettingsDialog::UpdateTimerLabel(const u32 id)
+{
+	switch (id)
+	{
+	case id_pad_lstick_left: b_left_lstick->setText(QString::number(m_seconds + 47)); break;
```

Comment:
Consider using a lambda for this update label.  I have no idea what +47 means, and a lambda would let you comment on its meaning.

Add this to the top of the function...

// Lambda used to update label. The 47 is magical.
auto UpdateLabel = [=](QPushButton* target) {
    target->setText(QString::number(m_seconds + 47);
};

And use it like this:

case id_pad_lstick_left: UpdateLabel(b_left_lstick); break;